src/org/opendatakit/briefcase/pull/aggregate/AggregateCursor.java

```diff
@@ -0,0 +1,168 @@
+ * date of a submission stored in Aggregate's database, not to be mistaken with
+ * the completion or submission dates, which can be different.
+ * <p>
+ * The submission instanceID of a Cursor ({@link #lastReturnedValue} field) is used
+ * to further filter the contents of a submission instanceID page when the existing
+ * submissions from the provided date don't fit in the same page.
+ */
+public class AggregateCursor implements Cursor<AggregateCursor> {
+  /**
+   * This date is used only to compare Cursors that might have an empty value in lastUpdate
+   */
+  private static final OffsetDateTime SOME_OLD_DATE = OffsetDateTime.parse("2010-01-01T00:00:00.000Z");
+  // TODO v2.0 Use a better name, like xml
+  private final String value;
+  private final Optional<OffsetDateTime> lastUpdate;
+  private final Optional<String> lastReturnedValue;
+
+  private AggregateCursor(String value, Optional<OffsetDateTime> lastUpdate, Optional<String> lastReturnedValue) {
+    this.value = value;
+    this.lastUpdate = lastUpdate;
+    this.lastReturnedValue = lastReturnedValue;
+  }
+
+  public static AggregateCursor empty() {
+    return new AggregateCursor("", Optional.empty(), Optional.empty());
+  }
+
+  /**
+   * Parses the provided cursor xml document and returns a new Cursor instance.
+   */
+  public static Cursor from(String cursorXml) {
+    if (cursorXml.isEmpty()
+        || cursorXml.equals("0") // Ona compatibility
+    )
+      return AggregateCursor.empty();
+
+    XmlElement root = XmlElement.from(cursorXml);
+
+    Optional<OffsetDateTime> lastUpdate = root
+        .findElement("attributeValue")
+        .flatMap(XmlElement::maybeValue)
+        .map(Iso8601Helpers::parseDateTime);
+
+    Optional<String> lastReturnedValue = root
+        .findElement("uriLastReturnedValue")
+        .flatMap(XmlElement::maybeValue);
+
+    return new AggregateCursor(cursorXml, lastUpdate, lastReturnedValue);
+  }
+
+  /**
+   * Returns a synthetic Cursor instance with the provided values.
+   */
+  public static AggregateCursor of(OffsetDateTime lastUpdate, String lastReturnedValue) {
+    String cursorXml = String.format("<cursor xmlns=\"http://www.opendatakit.org/cursor\">" +
+            "<attributeName>_LAST_UPDATE_DATE</attributeName>" +
+            "<attributeValue>%s</attributeValue>" +
+            "<uriLastReturnedValue>%s</uriLastReturnedValue>" +
+            "<isForwardCursor>true</isForwardCursor>" +
+            "</cursor>",
+        lastUpdate.format(ISO_OFFSET_DATE_TIME),
+        lastReturnedValue
+    );
+    return new AggregateCursor(cursorXml, Optional.of(lastUpdate), Optional.of(lastReturnedValue));
+  }
+
+  /**
+   * Returns a synthetic Cursor instance with the provided values.
+   */
+  public static AggregateCursor of(LocalDate date) {
+    OffsetDateTime lastUpdate = date.atStartOfDay().atOffset(ZoneOffset.UTC);
+    String cursorXml = String.format("<cursor xmlns=\"http://www.opendatakit.org/cursor\">" +
```

Comment:
Could we remove this duplication, please?
<img width="2476" alt="Screen Shot 2019-06-30 at 11 02 01 AM" src="https://user-images.githubusercontent.com/55099/60400303-cd7b8e80-9b26-11e9-9fd2-59b654ed4076.png">
```
  /**
   * Returns a synthetic Cursor instance with the provided values.
   */
  public static AggregateCursor of(OffsetDateTime lastUpdate, String lastReturnedValue) {
    String cursorXml = getCursorXml(lastUpdate, lastReturnedValue);
    return new AggregateCursor(cursorXml, Optional.of(lastUpdate), Optional.of(lastReturnedValue));
  }

  /**
   * Returns a synthetic Cursor instance with the provided values.
   */
  public static AggregateCursor of(LocalDate date) {
    OffsetDateTime lastUpdate = date.atStartOfDay().atOffset(ZoneOffset.UTC);
    String cursorXml = getCursorXml(lastUpdate, "");
    return new AggregateCursor(cursorXml, Optional.of(lastUpdate), Optional.empty());
  }

  private static String getCursorXml(OffsetDateTime lastUpdate, String lastReturnedValue) {
    return String.format("<cursor xmlns=\"http://www.opendatakit.org/cursor\">" +
            "<attributeName>_LAST_UPDATE_DATE</attributeName>" +
            "<attributeValue>%s</attributeValue>" +
            "<uriLastReturnedValue>%s</uriLastReturnedValue>" +
            "<isForwardCursor>true</isForwardCursor>" +
            "</cursor>",
        lastUpdate.format(ISO_OFFSET_DATE_TIME),
        lastReturnedValue);
  }
```

Reply:
We can't remove duplication exactly as you propose, but we can make it better. It's key we have a `<tag/>`, as opposed to the `<tag></tag>` you're proposing. Don't ask me why but I've noticed different behavior with `<tag/>`.

I'll work in the extraction as you've laid it out, but with that detail on mind.